src/platform/packages/shared/kbn-discover-utils/src/utils/get_message_field_with_fallbacks.ts

```diff
@@ -7,19 +7,35 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
+import { unescape } from 'lodash';
 import { fieldConstants } from '..';
 import { LogDocumentOverview } from '../types';
 
-export const getMessageFieldWithFallbacks = (doc: LogDocumentOverview) => {
+export const getMessageFieldWithFallbacks = (
+  doc: LogDocumentOverview,
+  { includeFormattedValue = false }: { includeFormattedValue?: boolean } = {}
+) => {
   const rankingOrder = [
     fieldConstants.MESSAGE_FIELD,
     fieldConstants.ERROR_MESSAGE_FIELD,
     fieldConstants.EVENT_ORIGINAL_FIELD,
   ] as const;
 
   for (const rank of rankingOrder) {
-    if (doc[rank] !== undefined && doc[rank] !== null) {
-      return { field: rank, value: doc[rank] };
+    const value = doc[rank];
+
+    if (value !== undefined && value !== null) {
+      let formattedValue: string | undefined;
+
+      if (includeFormattedValue) {
+        try {
+          formattedValue = JSON.stringify(JSON.parse(unescape(value)), null, 2);
```

Comment:
Just wanted to clarify here: This will return a `formattedValue` for things that are not JSON objects (such as `true` or `null` or `"hello"`). In my testing this seemed to work fine, but I just wanted to clarify that this is the desired behavior.

Reply:
Hmm good point... It wasn't intentional but I think it's ok since at worst they get some highlighting in the code block. If there's a concern about this we can update it to add an additional check that the parsed value is an object or array.